### PR TITLE
Version scheme change: 0.5.0 -> 5.0.0

### LINKED ISSUE #6073: Switch to SemVer
I've been following this project for a long time, and I saw one bad tendency, presented in many aspects: retrograde development model.
I thought, MT is using Semantic Versioning? Maybe, I'm wrong, but let's look in **Semantic Versioning Specs**, item 4:

> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

(http://semver.org/#spec-item-4)

Here goes the main question: why the project leads are so afraid to break backwards compatibility (taking into account we're at 0.x)? Sure, there are many mods based on "stable releases", but isn't it the time to move forward, especially at this development stage?

(And I'm sorry for another "holywar" thread.)

### PR DESCRIPTION
**Pull request by nerzhul**
Fix #6073 

EDIT by paramat: Core dev support is recorded here https://github.com/minetest/minetest/issues/6073#issuecomment-396597034 4 approve, 4 neutral.
Approvals also recorded here https://github.com/orgs/minetest/teams/engine/discussions/12

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,8 +15,8 @@ set(GCC_MINIMUM_VERSION "4.8")
 set(CLANG_MINIMUM_VERSION "3.4")
 
 # Also remember to set PROTOCOL_VERSION in network/networkprotocol.h when releasing
-set(VERSION_MAJOR 0)
-set(VERSION_MINOR 5)
+set(VERSION_MAJOR 5)
+set(VERSION_MINOR 0
```
Comment (ClobberXD) [inline CMakeLists.txt]: Missing a `)`

Comment (nerzhul) [reply CMakeLists.txt]: oops :)

**Comment by paramat** [comment]
Also needs the end of README.md updating to describe the new version scheme.

**Comment by nerzhul** [comment]
@paramat done

File: README.md [hunk 2]
```diff
@@ -431,9 +431,9 @@ This is how we build Windows releases.
 Version scheme
 --------------
 
-Minetest doesn't follow semver. Instead, we do something roughly similar to 0.major.minor.
+Minetest follows partially semver.
```
Comment (paramat) [inline README.md]: Better English is:
'Minetest partially follows semver.'

Comment (nerzhul) [reply README.md]: yeah, i'm not very up this morning :)

Comment (paramat) [reply README.md]: Maybe also specify we use major.minor.patch?
Maybe inform when we started using M.M.P. and that we used 0.M.M up to 0.4? This text specifies what we used in the past too: "Prior to that, we used oldversion-dev.".

**Comment by paramat** [comment]
Maybe something like:
"Since 5.0.0 we loosely follow SemVer by using major.minor.patch. Prior to that we used 0.major.minor."

**Comment by paramat** [comment]
:+1:

**Comment by Wuzzy2** [comment]
No, just no. Either you fully commit SemVer or you don't.
In this case, you do not use SemVer, but rolled out your own versioning scheme.
This is fine, just be clear about it.

You could simply not mention SemVer at all, or you might just say that Minetest is following your own scheme which is similar to SemVer.

But don't say “we are partially following SemVer”, that's just dumb.

**Comment by sfan5** [comment]
Agree with this. The sentence "Semver was adopted after 0.4.X series" is especially unclear.

**Comment by nerzhul** [comment]
feel free to open a PR to fix it. Personally i just wanted to drop the initial 0, at least this part is done

**Comment by paramat** [comment]
'Partially following' is not the ideal wording, and the text is a little awkward and messy too. It can be cleaned up later by a better English writer (no offense) :)

**Pull request merged**